IB Diploma/student_manage.php

```diff
@@ -32,33 +35,79 @@
         $page->breadcrumbs
         ->add(__('Manage Student Enrolment'));
     
-    echo '<p>';
-    echo 'This page only displays students enroled in the current school year.';
-    echo '</p>';
-
     if (isset($_GET['return'])) {
         returnProcess($guid, $_GET['return'], null, null);
     }
-        $data = array('gibbonSchoolYearID' => $_SESSION[$guid]['gibbonSchoolYearID']);
-        $sql = "SELECT ibDiplomaStudentID, student.surname, student.preferredName, start.name AS start, end.name AS end, gibbonYearGroup.nameShort AS yearGroup, gibbonRollGroup.nameShort AS rollGroup, gibbonPersonIDCASAdvisor, advisor.surname AS advisorSurname, advisor.preferredName as advisorPreferredName FROM ibDiplomaStudent JOIN gibbonPerson AS student ON (ibDiplomaStudent.gibbonPersonID=student.gibbonPersonID) JOIN gibbonStudentEnrolment ON (ibDiplomaStudent.gibbonPersonID=gibbonStudentEnrolment.gibbonPersonID) LEFT JOIN gibbonSchoolYear AS start ON (start.gibbonSchoolYearID=ibDiplomaStudent.gibbonSchoolYearIDStart) LEFT JOIN gibbonSchoolYear AS end ON (end.gibbonSchoolYearID=ibDiplomaStudent.gibbonSchoolYearIDEnd) LEFT JOIN gibbonYearGroup ON (gibbonStudentEnrolment.gibbonYearGroupID=gibbonYearGroup.gibbonYearGroupID) LEFT JOIN gibbonRollGroup ON (gibbonStudentEnrolment.gibbonRollGroupID=gibbonRollGroup.gibbonRollGroupID) RIGHT JOIN gibbonPerson AS advisor ON (ibDiplomaStudent.gibbonPersonIDCASAdvisor = advisor.gibbonPersonID) WHERE gibbonStudentEnrolment.gibbonSchoolYearID=:gibbonSchoolYearID AND student.status='Full' ORDER BY start.sequenceNumber DESC, surname, preferredName";
-        $result = $pdo->select($sql, $data)->toDataSet();
     
-    $table = DataTable::create('casStudentManage')->withData($result);
-    $table->addHeaderAction('add')->setURL('/modules/'.$_SESSION[$guid]['module'].'/student_manage_add.php');
+            
+        $CASStudentGateway = $container->get(CASStudentGateway::class);
+        $userGateway = $container->get(UserGateway::class);
+        $gibbonSchoolYearID = $gibbon->session->get('gibbonSchoolYearID');
+        $gibbonSchoolYearSequenceNumber = $gibbon->session->get('gibbonSchoolYearSequenceNumber');
+        $gibbonPersonID = $gibbon->session->get('gibbonPersonID');
+        
+        $gibbonRollGroupID = $_GET['gibbonRollGroupID'] ?? NULL;
+        $criteria = $CASStudentGateway
+            ->newQueryCriteria()
```

Comment:
Be sure to add `true` in here as the parameter if you'd like to use the default pagination size. Otherwise you'll always get the full result set.

Reply:
Good catch, I'd intentionally set it to not paginate on the staff table and must have forgotten to enable it on the student tables.